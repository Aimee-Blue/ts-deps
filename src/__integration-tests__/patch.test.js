// @ts-check
import {
  ROOT,
  emptyDirSafe,
  spawnAndCheck,
  sortedDirectoryContents,
} from './helpers'
import { join } from 'path'
import { ensureDir, copy } from 'fs-extra'
const { patch } = require('../scripts')

jest.setTimeout(2 * 60 * 1000)

describe('patch', () => {
  describe('given fresh directory after init', () => {
    const initDir = join(ROOT, 'integration-test-patch')

    beforeAll(async () => {
      await emptyDirSafe(initDir)
      await ensureDir(initDir)

      const sourceLocation = join(ROOT, 'test-sources', 'init-full')

      await copy(sourceLocation, initDir, {
        overwrite: true,
      })

      spawnAndCheck('npm', ['init', '-y'], {
        cwd: initDir,
      })

      await patch({
        forceOverwrites: true,
        cwd: initDir,
      })
    })

    it('should output correct files', async () => {
      const contents = await sortedDirectoryContents(initDir, [
        '**',
        '!node_modules/**',
      ])

      expect(contents).toMatchSnapshot()
    })
  })
})
