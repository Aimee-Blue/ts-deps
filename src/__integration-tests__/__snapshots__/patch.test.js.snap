// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`patch given fresh directory after init should output correct files 1`] = `
Array [
  ".eslintignore",
  ".eslintrc.js",
  ".gitignore",
  ".yarnrc",
  "babel.config.js",
  "commitlint.config.js",
  "jest.config.integration.js",
  "jest.config.js",
  "package.json",
  "prettier.config.js",
  "release.config.js",
  "scripts/",
  "scripts/build.js",
  "scripts/clean.js",
  "scripts/combineCoverage.js",
  "scripts/release.js",
  "src/",
  "src/__tests__/",
  "src/__tests__/max.test.ts",
  "src/__tests__/min.test.js",
  "src/__tests__/test.d.ts",
  "src/index.ts",
  "src/max/",
  "src/max/index.d.ts",
  "src/max/index.js",
  "src/max/max.ts",
  "src/min/",
  "src/min/index.d.ts",
  "src/min/index.js",
  "src/min/min.js",
  "src/other.d.ts",
  "src/other.js",
  "ts-deps.config.js",
  "tsconfig.declarations.json",
  "tsconfig.json",
]
`;
